src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb

```diff
@@ -8,11 +8,14 @@ Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
 Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
 
 Namespace Microsoft.CodeAnalysis.VisualBasic
+    Partial Friend MustInherit Class StateMachineRewriter(Of TProxy)
 
-    Friend Module StateMachineStates
-        Public FinishedStateMachine As Integer = -2
-        Public NotStartedStateMachine As Integer = -1
-        Public FirstUnusedState As Integer = 0
-    End Module
+        Friend Enum StateMachineStates As Integer
```

Comment:
>StateMachineStates [](start = 20, length = 18)

It feels wrong to use an enum for these values, as the values that state machine states take are not restricted to these three values. Typical state machines have states with the value 1, 2, 3, etc.  It is just that those values and the part of the state machine they refer to are computed on the fly.